packages/peregrine/lib/Apollo/clearCustomerDataFromCache.js

```diff
@@ -1,14 +1,26 @@
-import { deleteCacheEntry } from './deleteCacheEntry';
-
 /**
- * Deletes all references to Customer from the apollo cache including entries
- * that start with "$" which were automatically created by Apollo InMemoryCache.
- * By coincidence this rule additionally clears CustomerAddress entries, but
- * we'll need to keep this in mind by adding additional patterns as MyAccount
- * features are completed.
+ * Deletes all references to Customer from the apollo cache.
+ * Related queries that have reference to the customer are also deleted
+ * through the cascade. Note, however, that all secondary references must
+ * be deleted in order for garbage collection to do its job.
  *
  * @param {ApolloClient} client
  */
 export const clearCustomerDataFromCache = async client => {
-    await deleteCacheEntry(client, key => key.match(/^\$?Customer/i));
+    // Cached data
+    client.cache.evict({ id: 'Customer' });
+    // Remove createCustomerAddress which references newly created addresses, avoiding gc
+    client.cache.evict({
+        id: 'ROOT_MUTATION',
```

Comment:
While we are changing caching behaviour just a heads up that a future version of apollo will automatically remove ROOT_MUTATION https://github.com/apollographql/apollo-client/blob/main/CHANGELOG.md#potentially-disruptive-changes - not sure if this is something we rely on.

Reply:
This is great news, and would mean we can remove all the `@connection(key: "createCustomerAddress")` overrides we do to prevent sensitive data from leaking.

On that note, why was this needed @JustinConabreeAbsolunet? I see two operations that use the `createCustomerAddress` mutation, and they both have connection keys. Is this just being extra safe and removing any reference to "customer", even if it's not PII?

Reply:
@tjwiebell Without this, if we create a new address and then sign out, the new CustomerAddress is kept in cache because the createCustomerAddress contains a ref to it. Since Apollo's garbage collection only deletes entries that have no other references (unreachable), we have to manually remove this in order for garbage collection to work.

To Fooman's point though, this can be removed once we move to V3.4 of Apollo since ROOT_MUTATION will be cleared automatically.